File: client/mobile/lib/services/review_service.dart
import 'api_client.dart';

class ReviewService {
  final ApiClient _client = ApiClient();

  Future<Map<String, dynamic>> submitReview({
    required String bookingId,
    required String entityType,
    required String entityId,
    required int rating,
    String? comment,
  }) async {
    try {
      final response = await _client.dio.post(
        '/reviews',
        data: {
          'booking_id': bookingId,
          'entity_type': entityType,
          'entity_id': entityId,
          'rating': rating,
          'comment': comment,
        },
      );
      return response.data['data'] as Map<String, dynamic>? ?? {};
    } catch (e) {
      throw _client.extractErrorMessage(e);
    }
  }

  Future<List<dynamic>> getReviewsForEntity(String entityType, String entityId) async {
    try {
      final response = await _client.dio.get('/reviews/$entityType/$entityId');
      return response.data['data'] as List<dynamic>? ?? [];
    } catch (e) {
      return [];
    }
  }
}

File: client/web/src/services/reviewService.js
import { api, extractErrorMessage } from './api';

export const reviewService = {
  /**
   * List reviews for an entity (provider, business, organization)
   */
  async listReviews(entityType, entityId) {
    try {
      const response = await api.get(`/reviews/${entityType}/${entityId}`);
      return response.data.data || [];
    } catch (err) {
      throw new Error(extractErrorMessage(err));
    }
  },

  /**
   * List authenticated user's submitted reviews
   */
  async getMyReviews() {
    try {
      const response = await api.get('/reviews/me');
      return response.data.data || [];
    } catch (err) {
      throw new Error(extractErrorMessage(err));
    }
  },

  /**
   * Create a new review for a completed booking
   * @param {Object} data { booking_id, entity_type, entity_id, rating, comment }
   */
  async createReview(data) {
    try {
      const response = await api.post('/reviews', data);
      return response.data.data;
    } catch (err) {
      throw new Error(extractErrorMessage(err));
    }
  },

  /**
   * Update an existing review
   */
  async updateReview(id, data) {
    try {
      const response = await api.put(`/reviews/${id}`, data);
      return response.data.data;
    } catch (err) {
      throw new Error(extractErrorMessage(err));
    }
  },
};
